File: budget_calculator.py
# eg 7th budget calculator

income = float(input("What is your monthly income: "))
rent = float(input("What is your monthly rent/mortgage: "))
utilities = float(input("What is your monthly utilities: "))
groceries = float(input("What is your monthly groceries: "))
transportation = float(input("What is your monthly transportation: "))

rent_pct = rent / income * 100
utilities_pct = utilities / income * 100
groceries_pct = groceries / income * 100
transportation_pct = transportation / income * 100

save = income * 0.10
spending_money = income - (rent + utilities + groceries + transportation + save)
print(f"\nYour rent is ${rent:.2f} and that is {rent_pct:.0f}% of your income.")
print(f"Your utilities are ${utilities:.2f} and that is {utilities_pct:.0f}% of your income.")
print(f"Your groceries are ${groceries:.2f} and that is {groceries_pct:.0f}% of your income.")
print(f"Your transportation is ${transportation:.2f} and that is {transportation_pct:.0f}% of your income.")
print(f"\nYou should save ${save:.2f} a month, that is 10% of your income.")
print(f"\nYou have ${spending_money:.2f} of spending money.")

File: c_programming/notes/budget-calculator.c
// eg 7th budget calculaotr

#include <stdio.h>

int main() {
    float income, rent, utilities, groceries, transportation;
    float total_expenses, savings, leftover;

    
    printf("Enter your monthly income: $");
    scanf("%f", &income);

    printf("Enter your monthly Rent/Mortgage: $");
    scanf("%f", &rent);

    printf("Enter your monthly Utilities: $");
    scanf("%f", &utilities);

    printf("Enter your monthly Groceries: $");
    scanf("%f", &groceries);

    printf("Enter your monthly Transportation: $");
    scanf("%f", &transportation);

    
    total_expenses = rent + utilities + groceries + transportation;
    savings = income * 0.20; 
    leftover = income - total_expenses - savings;

    
    printf("\n--- Monthly Budget Summary ---\n");
    printf("Income: $%.2f\n", income);
    printf("Rent/Mortgage: $%.2f (%.2f%%)\n", rent, (rent / income) * 100);
    printf("Utilities: $%.2f (%.2f%%)\n", utilities, (utilities / income) * 100);
    printf("Groceries: $%.2f (%.2f%%)\n", groceries, (groceries / income) * 100);
    printf("Transportation: $%.2f (%.2f%%)\n", transportation, (transportation / income) * 100);
    printf("Recommended Savings: $%.2f (%.2f%%)\n", savings, (savings / income) * 100);
    printf("Leftover to spend: $%.2f (%.2f%%)\n", leftover, (leftover / income) * 100);

    return 0;
}